src/pages/Debug/DebugDetails.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useState} from 'react';
+import type {OnyxKey} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import Button from '@components/Button';
+import ScrollView from '@components/ScrollView';
+import Text from '@components/Text';
+import TextInput from '@components/TextInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+
+type DebugDetailsProps = {
+    data: Record<string, unknown>;
+    onyxKey: OnyxKey;
+    isCollection?: boolean;
+    idSelector?: (data: Record<string, unknown>) => string;
+};
+
+function DebugDetails({data, onyxKey, isCollection = false, idSelector = () => ''}: DebugDetailsProps) {
```

Comment:
```suggestion
function isObject(value: unknown): value is Record<string, unknown> {
    return typeof value === 'object';
}

function DebugDetails({data}: DebugDetailsProps) {
```

Let's create this `isObject` helper function.